### ISSUE TITLE
Failed to update the elementum library in a directory not normally used.

### ISSUE THREAD

**Issue by susobaco** [elgatito/plugin.video.elementum#950]
Hi, I am using elementum 0.1.98 on Kodi Nexus on Trisquel Nabia (based on ubunto 22.04) on a x64 platform and python 3.8.

My elementum library configuration, is not the usual one, but is /home/kodi/library.

I have noticed that lately it does not update the library (I never had problems in this aspect).

My .kodi/userdata/sources.xml configuration:

> <sources>
    <programs>
        <default pathversion="1"></default>
    </programs>
    <video>
        <default pathversion="1"></default>
        <source>
            <name>tmp</name>
            <path pathversion="1">/tmp/</path>
            <allowsharing>true</allowsharing>
        </source>
        <source>
            <name>Series Elementum</name>
            <path pathversion="1">/home/kodi/Librería/Shows/</path>
            <allowsharing>true</allowsharing>
        </source>
        <source>
            <name>Movies Elementum</name>
            <path pathversion="1">/home/kodi/Librería/Movies/</path>
            <allowsharing>true</allowsharing>
        </source>
[bundle_2023_10_29_10_3.log](https://github.com/elgatito/plugin.video.elementum/files/13197673/bundle_2023_10_29_10_3.log)

I attach detailed log.

I think I have attached everything I need, if you need any more information, do not hesitate to ask me, I will try to send it within my possibilities.

Apologies for my clumsiness.

**Comment by antonsoroko** [comment]
@susobaco 
this is simple:
`2023-10-29 10:02:28.182 T:19430 warning <general>: [plugin.video.elementum]   TraktAuthorized: false,`

you need to authorize Trakt again. go to elementum settings -> Trakt -> authorize.
You need to do this from time to time. And elementum shows you a warning message every time you run it - something like "authorize Trakt again" :-)

**Comment by susobaco** [comment]
@antonsoroko But, I never had the synchronization to Trakt activated. I don't even have an account.

**Comment by antonsoroko** [comment]
@susobaco Then about what kind of "update" are you talking about?

**Comment by susobaco** [comment]
@antonsoroko Before, when a chapter of a series was broadcasted, the library was automatically updated. Now, I think that with one of the last updates of elemtum it stopped doing it. If I search for a series with the elementum search engine and I click on the merge section, the episodes that have been broadcasted appear.
I can try an older version to test. As soon as I have time, I will do it.

**Comment by antonsoroko** [comment]
okay. if you will be able to determine which version works for you - then we can try to see what code changes caused the issue.

**Comment by susobaco** [comment]
I have tried several versions of the plugin and it still does not update the library. Then searching, I noticed that the TVDB provider is broken, unmaintained, and I installed the "TVDB v4" script but the library is still not updated. In the series search the new chapters appear, but they are not "injected" in the series directory (in the elementum library). 
Elementum gets information from the series using TVDB?
Maybe the library update is failing because the TVDB plugin is out of date?

**Comment by antonsoroko** [comment]
Elementum uses its own internal code for Trakt + T**M**DB, not external Kodi's addons.
But I never used trakt without login, so I am not sure what you need to do to force elementum to write .strm/.nfo files into elementum's library directory.
**I guess you need to make sure that you added tv show from elementum to kodi manually, by using context menu.**

Because without login elementum can not sync your own trakt lists (like watchlist, collections and etc trakt-specific), but looks like you do not use trakt for tracking (as majority of users use it) but you use trakt only as search engine.

**Comment by antonsoroko** [comment]
@susobaco 
Elementum uses its own internal code for Trakt + T**M**DB, not external Kodi's addons.
But I never used trakt without login, so I am not sure what you need to do to force elementum to write .strm/.nfo files into elementum's library directory.
**I guess you need to make sure that you added tv show from elementum to kodi manually, by using context menu.**

Because without login elementum can not sync your own trakt lists (like watchlist, collections and etc trakt-specific), but looks like you do not use trakt for tracking (as majority of users use it) but you use trakt only as search engine.

Maybe @elgatito can add more clarification.

**Comment by susobaco** [comment]
I don't know if it has anything to do with the changes you describe in this forum affecting scrapers. 

https://forum.kodi.tv/showthread.php?tid=370489

**Comment by antonsoroko** [comment]
here is example of NFO file generated by elementum:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<tvshow>
	<uniqueid type="unknown" default="false">95557</uniqueid>
	<uniqueid type="elementum" default="false">95557</uniqueid>
	<uniqueid type="tmdb" default="true">95557</uniqueid>
	<uniqueid type="imdb" default="false">tt6741278</uniqueid>
	<uniqueid type="tvdb" default="false">368207</uniqueid>
</tvshow>
https://www.themoviedb.org/tv/95557
https://www.imdb.com/title/tt6741278/
https://www.thetvdb.com/?tab=series&id=368207&lid=7
```

as you can see it has IDs for all popular web sites. then appropriate Kodi addon will use this info to scrape real data from web sites.

i think you misconfigured your Kodi somehow. sync works for me and for other users.
my only advice: try to setup new kodi on separate device to see if it will work.

**Comment by elgatito** [comment]
@susobaco Do you have these folders:
`/home/kodi/Librería/Movies` and `/home/kodi/Librería/Shows/` ?

**Comment by elgatito** [comment]
@susobaco Can't see anything in the log. Can we do this:
- Change log level to Debug in Elementum's settings.
- In Elementum settings go to Kodi Library tab and click on "Update library" to run local update.

And share new log of course, maybe it will have more details.

**Comment by susobaco** [comment]
[kodi.log](https://github.com/elgatito/plugin.video.elementum/files/13719394/kodi.log)

I think that's what you're asking for, isn't it?

**Comment by elgatito** [comment]
@susobaco Next version will have a change regarding re-use of a folder.

**Comment by antonsoroko** [comment]
@elgatito is there a specific commit for that? just curious how it was fixed.

**Comment by elgatito** [comment]
@antonsoroko It should be the same as #949

**Comment by antonsoroko** [comment]
@elgatito but here user does not use `special://`, the only non-standard thing here is UTF-8 `í`, unless I am missing something.

**Comment by elgatito** [comment]
Well, yes, I thought it is `special...`.

@susobaco Can you try with v0.1.99 and post a new log if the issue still there?

**Comment by susobaco** [comment]
[kodi.log](https://github.com/elgatito/plugin.video.elementum/files/13798662/kodi.log)
It still doesn't work, when it reads to see if there are new episodes it doesn't find any, and if there are any
The following log is when I merge the series found in the elementum search engine.
[kodi.log](https://github.com/elgatito/plugin.video.elementum/files/13798673/kodi.log)

**Comment by antonsoroko** [comment]
@elgatito do I remember correctly that we skip sync if TraktAuthorized is not set? And here user decided not to login into trakt.
Or I am missing something.

**Comment by susobaco** [comment]
Hello, the latest versions I see that the problem has not been solved. Do you need any more information?
Cheers!

**Comment by antonsoroko** [comment]
@susobaco just for test - can you try to make your library directory without any special characters?
so instead of `Vídeos` - `Videos`, instead of `Librería` - `Libreria` and so on.

**Comment by susobaco** [comment]
Yes, I did, it was one of the first things I changed, although it always worked with “Library”. Now it is called “Libreria”.

**Comment by antonsoroko** [comment]
@susobaco in your last log file it is "Vídeos". So I mean change it to "Videos".

Anyway, you need to create a new log file on the specific moment when new episode was released but Elementum was not started yet. 
So then you start Elementum and it fetch that new episode and then we can see what happens in log. 

Any other log will be useless in my opinion.

**Comment by susobaco** [comment]
I attach two logs, the first one is from kodi at startup and the second one is when I do a search for the series that have already aired chapters and merge it with the library.
[kodi_0.log](https://githu
[kodi_1.log](https://github.com/elgatito/plugin.video.elementum/files/15420686/kodi_1.log)
b.com/elgatito/plugin.video.elementum/files/15420685/kodi_0.log)

**Comment by antonsoroko** [comment]
@susobaco first of all: **you have not enabled debug in Elementum itself.**  see https://elementum.surge.sh/contribute/log/ for a guide.

i see `LogLevel: 2,` which is "info" level.
btw, elgatito already asked you to enable debug in Elementum several messages above.

sorry, this mean that logs do not have useful info :-(

i do not even see if sync from trakt was started or not.

new log file is needed.

**also please wait at least 5 minutes after elementum started. and only then copy log.**

**Comment by antonsoroko** [comment]
internal note:

also, i still not sure about this question:
" @elgatito do I remember correctly that we skip sync if TraktAuthorized is not set? And here user decided not to login into trakt.
Maybe back in time the logic was different.
Or I am missing something. "

i still do not see in code how sync can work if user have not authorized trakt.
also here sync of a show that was manually added to kodi library, so show not in any list/collection.

**Comment by elgatito** [comment]
@susobaco Please, enable debug logging and make a new log.

**Comment by AndroG77** [comment]
Sadly I also have had this issue for the last year.
All the TV Shows added to the Kodi library from elementum don't update when there is a new episode, but they are available in Elementum, just not in the Kodi library.

**Comment by antonsoroko** [comment]
@AndroG77 did you authorize trakt in Elementum? 
I mean: Elementum settings -> Trakt -> authorize.
Theoretically it is not necessary (at least it used to be) to authorize trakt, but it might be the issue nowadays.

Also, we still need proper debug log file. See https://github.com/elgatito/plugin.video.elementum/issues/950#issuecomment-2129415366 for the guide how to make it. 

We can't to fix something without proper technical information.

**Comment by AndroG77** [comment]
I don't use Trakt and nothing is highlighted in Elementum regarding Trakt. So no authorisation for it and no username.
I will upload the log tomorrow because tonight there will be a new episode of a show and I expect it to arrive in Elementum and not be added to the kodi library.

**Comment by antonsoroko** [comment]
@AndroG77 please enable debug in Kodi and Elementum in advance (before test). Guide is above. 
Otherwise we will miss the "event" of "new episode was released".

**Comment by AndroG77** [comment]
So as expectd the Elementum has a new episode for a show but it doesn't get added to the Kodi library. When I manually trigger to scan the video library from Kodi nothing gets added. Here's the log:

[kodi.log](https://github.com/user-attachments/files/17160729/kodi.log)

**Comment by antonsoroko** [comment]
@AndroG77 what is the full name of episode? (Show name + season number + episode number)

**Comment by AndroG77** [comment]
The Lord of the Rings: The Rings of Power, Season 02, Episode 07

**Comment by elgatito** [comment]
@AndroG77 

1) Elementum cannot add something into Kodi library, it can only write `strm` files to the disk, and expecting you have added that folder as source for movies/shows. After each show update, when there are new episodes (and if you have not disabled that behavios through Elementum settings) Elementum will trigger Kodi library update. So if you configured library paths properly, if Elementum is creating new strm files - you will have automatic update of library and new episodes appearing.

2) Elementum checks for new episodes for list of shows for:
 -  "internal Elementum library", which is a database stored on the disk, that you can remove or cleanup, so that library will be wiped.
 - shows, added to your Trakt account. This does not depend on internal library.

So if you don't use Trakt for tracking your library - then your flow for understanding an error is this:
- Check that Elementum is creating new strm files
    - If not: check that Elementum has that show in internal library, by going to Elementum -> Shows -> In Elementum library (on the bottom of the list). If not in the list: Find it, get context menu, "Add to the library" again.
    - Check the folder that is set to Elementum library.
- If strm files are properly written but for getting into Kodi library - Check that folder is added as source for Kodi library. 

That is all what Elementum can do, later Shows/Movies are parsed by scrappers.

**Comment by AndroG77** [comment]
Ok, I checked it all (also on a Windows laptop, that has the same issue) and all the settings are still the same as they always have been. When I add a new show to the library from Elementum it gets added to the Kodi library folder I specified, only the new epidoes (strm file) will not get added anymore in the Kodi library. This issue has been going on for a year now and I didn't change any settings (same for the Windows laptop) so it seems a certain update of Elementum caused this issue. All the library settings are correct and the library path is added as the source for Kodi library, but for some reason Elemtum won't add the new episode strm file to the path I specified in Elementum for the Kodi library.
When I manually merge the show from Elementum to my library then it will add it, so it seems there is an issue in Elementum regarding the library integration in Kodi.

**Comment by elgatito** [comment]
@AndroG77 

Ok, do this:
- Enable Debug logging level in Elementum settings.
- Restart Kodi.
- Wait for 1 minute (don't do anything)
- Make a log file - https://elementum.surge.sh/contribute/log/

**Comment by AndroG77** [comment]
Hey, here's the log:

[kodi.log](https://github.com/user-attachments/files/17170421/kodi.log)

**Comment by AndroG77** [comment]
As you can see in the log it says:
 **2024-09-27 23:01:47.609 T:820     DEBUG <general>: VideoInfoScanner: Skipping dir '/storage/library/Shows/The Lord of the Rings The Rings of Power (2022)/' due to no change**
I don't know if this has something to do with it or if this maybe is only for changed or updated information on the show itself.
Thanks by the way for trying to figure it out.
(And, of course, thanks for creating Elementum in the first place and after years of having used it it still is the best addon for Kodi even though this really minor annoyance for the last year, but I'm confident this can be fixed)

**Comment by elgatito** [comment]
@AndroG77 

```
Library updated in ...
```

This is the line I was hoping to see. But I don't.
Will need to look there precisely.

You use Linux_armv7 on that device? 
I will probably make you a test bundle with more logging to understand why it does not want to make local updates.

**Comment by antonsoroko** [comment]
@elgatito i failed to find where elementum does sync of shows that were manually added (so they are not from lists/collection and thus no trakt auth is needed).
can you please give me a link to code? i tried many times to find it, but i am not good at that part of code.

**Comment by AndroG77** [comment]
Hey, yes I use it on a Raspberry Pi  4 with Libreelec

**Comment by susobaco** [comment]
Sorry for the long response time, health problems.
I attach a log, this machine is debian 12 amd64 . and I think I have put the configuration right. The series I should update is Beacon 23 and the library is in a directory with no special characters.
[kodi.log](https://github.com/user-attachments/files/17241002/kodi.log)

**Comment by elgatito** [comment]
@AndroG77 Linux Armv7 build: 
[plugin.video.elementum-v0.1.103-6-g79f7828.linux_armv7.24.zip](https://github.com/user-attachments/files/17241323/plugin.video.elementum-v0.1.103-6-g79f7828.linux_armv7.24.zip)

@susobaco Linux x64 build: 
[plugin.video.elementum-v0.1.103-6-g79f7828.linux_x64.24.zip](https://github.com/user-attachments/files/17241330/plugin.video.elementum-v0.1.103-6-g79f7828.linux_x64.24.zip)

Same task for both, install from zip (on top of existing addon, don't need to uninstall it).
Restart Kodi, wait for one minute and take a log file.

**Comment by susobaco** [comment]
I think the new plugin causes a sudden shutdown of kodi. Attached log
[kodi.log](https://github.com/user-attachments/files/17241903/kodi.log)

**Comment by elgatito** [comment]
> I think the new plugin causes a sudden shutdown of kodi. Attached log [kodi.log](https://github.com/user-attachments/files/17241903/kodi.log)

It does not even seem to try starting Elementum in the log. In your case Kodi quits?

**Comment by AndroG77** [comment]
Here's my log:

[kodi.log](https://github.com/user-attachments/files/17242369/kodi.log)

**Comment by elgatito** [comment]
> Here's my log:
> 
> [kodi.log](https://github.com/user-attachments/files/17242369/kodi.log)

@AndroG77 Thanks, did see part of needed data. Can you update with this version and make another log?

[plugin.video.elementum-v0.1.103-7-g8f9f74b.linux_armv7.26.zip](https://github.com/user-attachments/files/17242676/plugin.video.elementum-v0.1.103-7-g8f9f74b.linux_armv7.26.zip)

**Comment by elgatito** [comment]
Here is the updated test version for Linux Armv7
[plugin.video.elementum-v0.1.103-7-g8f9f74b.linux_armv7.27.zip](https://github.com/user-attachments/files/17242912/plugin.video.elementum-v0.1.103-7-g8f9f74b.linux_armv7.27.zip)

**Comment by AndroG77** [comment]
Here's the log:

[kodi.log](https://github.com/user-attachments/files/17242986/kodi.log)

**Comment by elgatito** [comment]
> > Here is the updated test version for Linux Armv7 [plugin.video.elementum-v0.1.103-7-g8f9f74b.linux_armv7.27.zip](https://github.com/user-attachments/files/17242912/plugin.video.elementum-v0.1.103-7-g8f9f74b.linux_armv7.27.zip)
> 
> Do I need to install this one and make an new log?

Yes, please, it has a fix for one of the issues.

**Comment by AndroG77** [comment]
OK, here's the log:

[kodi.log](https://github.com/user-attachments/files/17243101/kodi.log)

**Comment by AndroG77** [comment]
Just checked and it seems the issue is fixed. The TV shows now update again and Kodi has the latest episode!

Thanks a lot man!

**Comment by elgatito** [comment]
> Just checked and it seems the issue is fixed. The TV shows now update again and Kodi has the latest episode!
> 
> Thanks a lot man!

> 2024-10-03 12:52:43.102 T:942   WARNING <general>: [plugin.video.elementum] INFO  library      ▶ updateLibraryShows  [0mLibrary updated in 27.817823543s

Yes! Finally. You deserve a mention in release docs for tracking this down and making log files to find the real cause 🥇

**Comment by susobaco** [comment]
> It does not even seem to try starting Elementum in the log. In your case Kodi quits?

2024-10-03 13:10:10.172 T:278773   debug <general>: CPythonInvoker(19, /home/manteniment/.kodi/addons/plugin.video.thecrew/thecrew.py): instantiating addon using automatically obtained id of "plugin.video.thecrew" dependent on version 3.0.0 of the xbmc.python api
2024-10-03 13:10:10.196 T:278653   error <general>: WARNING:root:Failed getting kodi platform: 'ascii' codec can't decode byte 0xef in position 0: ordinal not in range(128)
                                                   Traceback (most recent call last):
                                                     File "/home/manteniment/.kodi/addons/repository.github/lib/platform_/kodi_platform.py", line 57, in dump_platform
                                                       return get_kodi_platform_from_log()
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/home/manteniment/.kodi/addons/repository.github/lib/platform_/kodi_platform.py", line 48, in get_kodi_platform_from_log
                                                       next(f)
                                                     File "/usr/lib/python3.11/encodings/ascii.py", line 26, in decode
                                                       return codecs.ascii_decode(input, self.errors)[0]
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                   UnicodeDecodeError: 'ascii' codec can't decode byte 0xef in position 0: ordinal not in range(128)
                                                   
2024-10-03 13:10:10.196 T:278653   debug <general>: .


############### END LOG FILE ################

############ END Kodi CRASH LOG #############

**Comment by susobaco** [comment]
Fixed in the version 0.1.105, thanks for your work.

**Comment by susobaco** [comment]
Fixed in the version 0.1.105, thanks for your work.

**Comment by AndroG77** [comment]
Thanks again guys!
